src/ddmd/mtype.d

```diff
@@ -7688,10 +7688,13 @@ extern (C++) abstract class TypeQualified : Type
                 error(loc, "'%s' is not defined, perhaps you need to import %s; ?", p, n);
             else
             {
-                auto id = new Identifier(p);
+                auto id = Identifier.idPool(p, strlen(p));
+                const(char)* q;
                 s = sc.search_correct(id);
                 if (s)
                     error(loc, "undefined identifier `%s`, did you mean %s `%s`?", p, s.kind(), s.toChars());
+                else if ((q = Scope.search_correct_C(id)) !is null)
```

Comment:
I think you can also use the following to narrow the scope `q` is visible in:
`else if (const(char)* q = Scope.search_correct_C(id))`